packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybridD1.hpp

```diff
@@ -0,0 +1,985 @@
+		       size_t vertex_list_size = 0,
+                       bool recolor=false){
+      
+      //set up types to be used by KokkosKernels
+      using KernelHandle =  KokkosKernels::Experimental::KokkosKernelsHandle
+          <offset_t, lno_t, lno_t, ExecutionSpace, MemorySpace, 
+           MemorySpace>;
+      using lno_row_view_t = Kokkos::View<offset_t*, Kokkos::Device<ExecutionSpace, MemorySpace>>;
+      using lno_nnz_view_t = Kokkos::View<lno_t*, Kokkos::Device<ExecutionSpace, MemorySpace>>;
+
+      KernelHandle kh;
+      
+      //pick which KokkosKernels algorithm to use. Typically based on max graph degree.
+      if(recolor){
+        kh.create_graph_coloring_handle(KokkosGraph::COLORING_VBBIT);
+      } else {
+        kh.create_graph_coloring_handle(KokkosGraph::COLORING_EB);  
+      }
+      
+      //vertex_list_size indicates whether we have provided a list of vertices to recolor.
+      //Only makes a difference if the algorithm to be used is VBBIT.
+      if(vertex_list_size != 0){
+        kh.get_graph_coloring_handle()->set_vertex_list(vertex_list,vertex_list_size);
+      }
+
+      kh.set_verbose(verbose);
+
+      //set the initial coloring of the kh.get_graph_coloring_handle() to be
+      //the data view from the femv.
+      auto femvColors = femv->template getLocalView<MemorySpace>();
+      auto  sv = subview(femvColors, Kokkos::ALL, 0);
+      kh.get_graph_coloring_handle()->set_vertex_colors(sv);
+      kh.get_graph_coloring_handle()->set_tictoc(verbose);
+      KokkosGraph::Experimental::graph_color_symbolic<KernelHandle, lno_row_view_t, lno_nnz_view_t>
+                                                     (&kh, nVtx, nVtx, offset_view, adjs_view);
+      numColors = kh.get_graph_coloring_handle()->get_num_colors();
+
+      if(verbose){
+        std::cout<<"\nKokkosKernels Coloring: "<<kh.get_graph_coloring_handle()->get_overall_coloring_time()<<" iterations: "<<kh.get_graph_coloring_handle()->get_num_phases()<<"\n\n";
+      }
+    }
+    
+    template <class ExecutionSpace, typename MemorySpace>
+    void detectConflicts(const size_t n_local, const size_t n_ghosts,
+		         Kokkos::View<offset_t*, Kokkos::Device<ExecutionSpace, MemorySpace>> dist_offsets,
+			 Kokkos::View<lno_t*, Kokkos::Device<ExecutionSpace, MemorySpace>> dist_adjs,
+			 Kokkos::View<int*, Kokkos::Device<ExecutionSpace, MemorySpace>> femv_colors,
+			 Kokkos::View<lno_t*,
+			              Kokkos::Device<ExecutionSpace, MemorySpace>,
+				      Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_send_atomic,
+		         Kokkos::View<size_t[1],
+			              Kokkos::Device<ExecutionSpace, MemorySpace>,
+				      Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_send_size_atomic,
+			 Kokkos::View<gno_t[1], Kokkos::Device<ExecutionSpace, MemorySpace>> recoloringSize,
+			 Kokkos::View<int*,
+			              Kokkos::Device<ExecutionSpace, MemorySpace>> rand,
```

Comment:
Why do you need to pass a rand vector around? I thought it would be simpler to just generate the pseudo-random numbers on the fly (as needed).